core/src/main/R/base/strwrap.R

```diff
@@ -191,3 +191,16 @@ function(x, y, style = c("table", "list"),
     }
     r
 }
+
+trimws <- function(x, which = c("both", "left", "right"),
+                   whitespace = "[ \t\r\n]")
+{
+    which <- match.arg(which)
+    mysub <- function(re, x) sub(re, "", x, perl = TRUE)
```

Comment:
Avoid definining an inline function here. That causes problems (currently) for Renjin's optimizer, and makes the code harder to read. You can just make the switch statement an argument of the sub function as the pattern is the only thing the changes.